src/NuGet.Server.Core/Infrastructure/ServerPackage.cs

```diff
@@ -249,5 +253,41 @@ private static List<PackageDependencySet> ParseDependencySet(string value)
 
             return Tuple.Create(id, versionSpec, targetFramework);
         }
+
+        private static bool IsPackageSemVer2(IPackage package)
```

Comment:
I think NuGet.Server should be separate. It depends on NuGet.Core (`IPackage`). Also, there are currently no dependencies between gallery and NuGet.Server repositories.

I think the sharing should be done amongst NuGet.org components, but not with NuGet.Server.